### ISSUE TITLE
Incorrect windows build instructions in README.md

### ISSUE THREAD

**Issue by paulotten** [uutils/coreutils#2045]
https://github.com/uutils/coreutils#cargo states

> Because the above command attempts to build utilities that only work on Unix-like platforms at the moment, to build on Windows, you must do the following:
> `# to keep debug information, compile without --release`
> `$ cargo build --release --no-default-features --features windows`

However `cargo build --release` works just fine for me on windows.

Is this an outdated section that should be removed from the readme?

**Comment by Arcterus** [comment]
This is outdated AFAIK.  Unix is no longer assumed by default (and thus needs to be manually specified just like Windows).

**Comment by paulotten** [comment]
> (and thus needs to be manually specified just like Windows).

I don't think so. `cargo build --release` works fine for me on WSL too.

**Comment by paulotten** [comment]
> I don't think we have any windows only utilities so on windows cargo build --release should work fine.

No, I seem to be wrong on that.

So `cargo build --release --features OS_TYPE_HERE` for every OS is advisable.

**Comment by rivy** [comment]
> TLDR; use `cargo build --features windows`, `cargo build --features macos`, or `cargo build --features unix` to build the most complete utility set for that respective platform.

The README is essentially correct but, yes, it could do with some updating; I'll create some changes tomorrow.

No need to read further unless you're interested in a deeper, more detailed explanation.

---

>Referring to the main coreutils *Cargo.toml* `[features]` section and associated commentary might be helpful in understanding the various available feature sets and how they interact.

Historically, I build the current feature set (see b7a3c4d8 and 57ef582b) as a flexible way to handle differences in platform capability to build/run the various utilities. When I refactored the features, I based them on a common core set ("feat_common_core") that runs on most targets and is configured as the feature default. By doing so, a simple `cargo build` will generate a binary containing that utility subset on any of the three major platforms (MacOS, Unix, and Windows). This makes the initial developer/user experience more pleasant and less confusing.

As noted, however, it only builds the limited subset of utilities which essentially build everywhere. Further feature sets build on "feat_common_core" to trigger including additional utilities which are more platform-specific.

```toml
# "feat_Tier1" == expanded set of utilities which can be built/run on the usual rust "Tier 1" target platforms  (ref: <https://forge.rust-lang.org/release/platform-support.html>)
feat_Tier1 = [
    "feat_common_core",
    #
    "arch",
    "hostname",
    "nproc",
    "sync",
    "touch",
    "whoami",
]
## (primary platforms) feature sets
# "feat_os_macos" == set of utilities which can be built/run on the MacOS platform
feat_os_macos = [
    "feat_os_unix", ## == a modern/usual *nix platform
]
# "feat_os_unix" == set of utilities which can be built/run on modern/usual *nix platforms
feat_os_unix = [
    "feat_Tier1",
    #
    "feat_require_crate_cpp",
    "feat_require_unix",
    "feat_require_unix_utmpx",
]
# "feat_os_windows" == set of utilities which can be built/run on modern/usual windows platforms
feat_os_windows = [
    "feat_Tier1", ## == "feat_os_windows_legacy" + "hostname"
]
```

Here, in an excerpt from the *Cargo.toml* file, you can see that "feat_Tier1" includes the common core set plus "arch", "hostname", "nproc", "sync", "touch", and "whoami". Windows ("feat_os_windows") is a Tier1, Unix has Tier1 plus utility sets with various requirements (documented and implemented later in the feature section), and that MacOS is just an alias of Unix.

Specific utilities can be built into the final 'coreutils' binary simply by mentioning them as a "feature"; eg, `cargo build --no-default-features --features "base32"` will build a `coreutils` binary which includes only the "base32" utility.

Note that, because of the unfortunate design decision to use the same namespace for features and dependencies, all feature names use the "feat_..." nomenclature to minimize risk of name collision with the utilities themselves. However, for convenience at the command line, I did add "macos", "windows", and "unix" as feature names which are aliases of "feat_os_macos", "feat_os_windows", and "feat_os_unix", respectively. These could potentially collide if any utility named "macos", "windows", or "unix" ever gets added to the utility set, but they were too convenient to avoid.
